orttraining/orttraining/training_api/include/onnxruntime_training_c_api.h

```diff
@@ -11,310 +11,360 @@ ORT_RUNTIME_CLASS(CheckpointState);  /// Type that holds the training states for
-  *
-  * \snippet{doc} snippets.dox OrtStatus Return Value
-  *
-  */
+   *
+   * This function allows users to get the learning rate for the training session. The current
+   * learning rate is maintained by the training session
+   *
+   * \param[in] sess The training session on which the learning rate needs to be set.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   */
   ORT_API2_STATUS(GetLearningRate, _Inout_ OrtTrainingSession* sess, _Out_ float* learning_rate);
 
   /** \brief Performs the weight updates for the trainable parameters using the optimizer model.
-  *
-  * This function performs the weight update step that updates the trainable parameters such that they
-  * take a step in the direction of their gradients. The optimizer step is performed based on the optimizer
-  * model that was provided to the training session.
-  * The updated parameters are stored inside the training session so that they can be used by the next
-  * TrainStep function call.
-  *
-  * \param[in] sess The training session which has working knowledge of the optimizer model.
-  * \param[in] run_options Run options for this eval step.
-  *
-  * \snippet{doc} snippets.dox OrtStatus Return Value
-  *
-  */
+   *
+   * This function performs the weight update step that updates the trainable parameters such that they
+   * take a step in the direction of their gradients. The optimizer step is performed based on the optimizer
+   * model that was provided to the training session.
+   * The updated parameters are stored inside the training session so that they can be used by the next
+   * TrainStep function call.
+   *
+   * \param[in] sess The training session which has working knowledge of the optimizer model.
+   * \param[in] run_options Run options for this eval step.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   */
   ORT_API2_STATUS(OptimizerStep, _Inout_ OrtTrainingSession* sess,
                   _In_opt_ const OrtRunOptions* run_options);
 
   /** \brief Registers the use of the Linear learning rate scheduler for the training session.
-  *
-  * Register a Linear learning rate scheduler with the given
-  * learning rate scheduler parameters. Optionally specify the initial learning rate
-  * that should be used with this learning rate scheduler and training session.
-  *
-  * \param[in] sess The training session that should use the linear learning rate scheduler.
-  * \param[in] warmup_step_count Warmup steps for LR warmup.
-  * \param[in] total_step_count Total step count.
-  * \param[in] initial_lr The initial learning rate to be used by the training session.
-  *
-  * \snippet{doc} snippets.dox OrtStatus Return Value
-  *
-  */
+   *
+   * Register a Linear learning rate scheduler with the given learning rate scheduler parameters.
+   * Users must specify the initial learning rate that should be used with this learning rate scheduler
```

Comment:
Internally initial_lr is treated as optional:
Status TrainingSession::RegisterScheduler(const std::function<std::unique_ptr<LRSchedulerBase>std::shared_ptr<Optimizer>)>& get_scheduler,                            **std::optional<float>** initial_lr);

We should change this too or another option is to change the C API to accept float* to keep initial_lr as optional.